### PR TITLE
[CU-2wp5e0r] Add running of tests with coverage and fix cspell warnings in nix CI pipeline

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2wp5e0r

### CODE REVIEW

File: .config/spellcheck.dic [hunk 1]
```diff
@@ -230,3 +230,928 @@ XCM
 XCMP
 XCVM
 yzhang
+Aave
+absentations
+accesss
+accound
+accross
+accue
```
Comment (benluelo) [inline .config/spellcheck.dic:238]:
```suggestion
```

File: .config/spellcheck.dic [hunk 2]
```diff
@@ -230,3 +230,928 @@ XCM
...
+ACENT
+ACENTS
+acknowlegdement
+acks
+Acks
+actaul
```
Comment (benluelo) [inline .config/spellcheck.dic:244]:
```suggestion
```

File: .config/spellcheck.dic [hunk 3]
```diff
@@ -230,3 +230,928 @@ XCM
...
+actioned
+actualy
+acutal
```
Comment (benluelo) [inline .config/spellcheck.dic:247]:
```suggestion
```

File: .config/spellcheck.dic [hunk 4]
```diff
@@ -230,3 +230,928 @@ XCM
...
+adda
+addendums
+addn
+addtional
+adoc
+ADOC
+Adorment
+AETH
+agains
```
Comment (benluelo) [inline .config/spellcheck.dic:256]:
```suggestion
```

File: .config/spellcheck.dic [hunk 5]
```diff
@@ -230,3 +230,928 @@ XCM
...
+Aggarwal
+Aggr
+Airdropper
+aitlahcen
+aitlance
+Akala
+alices
+alloacte
+alread
```
Comment (benluelo) [inline .config/spellcheck.dic:265]:
```suggestion
```

File: .config/spellcheck.dic [hunk 6]
```diff
@@ -230,3 +230,928 @@ XCM
...
+AMM’s
+Amnt
+amout
+Amout
+analysing
+Anchorable
+Anohter
```
Comment (benluelo) [inline .config/spellcheck.dic:272]:
```suggestion
```

File: .config/spellcheck.dic [hunk 7]
```diff
@@ -230,3 +230,928 @@ XCM
...
+blockbuilder
+blocktime
+Boba
+Bondings
+bondoffers
+bootnode
+Bootsrapping
+borrowable
+Borrowable
+borrwers
```
Comment (benluelo) [inline .config/spellcheck.dic:331]:
```suggestion
```

File: .config/spellcheck.dic [hunk 8]
```diff
@@ -230,3 +230,928 @@ XCM
...
+Brainjar
+BTCUSD
+buildpackage
+Burghi
+bussiness
+Buterin
+bytecode
+calcaulate
+caluclate
+cancelation
```
Comment (benluelo) [inline .config/spellcheck.dic:342]:
```suggestion
```

File: .config/spellcheck.dic [hunk 9]
```diff
@@ -230,3 +230,928 @@ XCM
...
+coeff
+Coeff
+Cofiguration
+collaed
+collapsable
+collateralisation
+collateralized
+Collateralized
+collaterlized
+colllection
```
Comment (benluelo) [inline .config/spellcheck.dic:371]:
```suggestion
```

File: .config/spellcheck.dic [hunk 10]
```diff
@@ -230,3 +230,928 @@ XCM
...
+cooloff
+Cooloff
+corellate
+coreutils
+correspondes
+Cosm
+cosmin
+cosmwasm
+Cosmwasm
+Couldnt
```
Comment (benluelo) [inline .config/spellcheck.dic:411]:
```suggestion
```

File: .config/spellcheck.dic [hunk 11]
```diff
@@ -230,3 +230,928 @@ XCM
...
+continoty
+Contribor
+controlorigin
+cooldown
+Cooldown
+cooldowns
+Cooldowns
+cooloff
+Cooloff
+corellate
```
Comment (benluelo) [inline .config/spellcheck.dic:404]:
```suggestion
```

File: .config/spellcheck.dic [hunk 12]
```diff
@@ -230,3 +230,928 @@ XCM
...
+coreutils
+correspondes
+Cosm
+cosmin
+cosmwasm
+Cosmwasm
+Couldnt
+covderts
+craete
+creater
```
Comment (benluelo) [inline .config/spellcheck.dic:414]:
```suggestion
```

File: .config/spellcheck.dic [hunk 13]
```diff
@@ -230,3 +230,928 @@ XCM
...
+crev
+Crocswap
+CROWDLOANREWARDS
+crowdloans
+Crowdloans
+crreate
```
Comment (benluelo) [inline .config/spellcheck.dic:420]:
```suggestion
```

File: .config/spellcheck.dic [hunk 14]
```diff
@@ -230,3 +230,928 @@ XCM
...
+customizability
+cypherpunk
+Darkpools
+datatypes
+datetime
+Decentralised
+DECENTRALISED
+decrementor
+Decrementor
+defaultt
```
Comment (benluelo) [inline .config/spellcheck.dic:439]:
```suggestion
```

File: .config/spellcheck.dic [hunk 15]
```diff
@@ -230,3 +230,928 @@ XCM
...
+denoms
+deopsited
+dependancy
+depositable
+Depositable
+depositn
+Deque
+Deregistration
+descript
+Descrition
```
Comment (benluelo) [inline .config/spellcheck.dic:450]:
```suggestion
```

File: .config/spellcheck.dic [hunk 16]
```diff
@@ -230,3 +230,928 @@ XCM
...
+desugared
+DHMS
+dicuss
+distro
+Dlab
+Dockerized
+Doesnt
```
Comment (benluelo) [inline .config/spellcheck.dic:457]:
```suggestion
```

File: .config/spellcheck.dic [hunk 17]
```diff
@@ -230,3 +230,928 @@ XCM
...
+existant
+expireable
+Exstrinsic
+externatlities
+Extesions
+extirnsic
+Extrinsiscs
+extrsinsic
+failes
+feautes
```
Comment (benluelo) [inline .config/spellcheck.dic:501]:
```suggestion
```

File: .config/spellcheck.dic [hunk 18]
```diff
@@ -230,3 +230,928 @@ XCM
...
+drawio
+DTCC
+duplicative
+dustable
+edgecase
+edgeware
+editting
+Editting
+embbed
+emphasising
```
Comment (benluelo) [inline .config/spellcheck.dic:473]:
```suggestion
```

File: .config/spellcheck.dic [hunk 19]
```diff
@@ -230,3 +230,928 @@ XCM
...
+Encodable
+endef
+endrosed
+endwhile
+enigne
```
Comment (benluelo) [inline .config/spellcheck.dic:478]:
```suggestion
```

File: .config/spellcheck.dic [hunk 20]
```diff
@@ -230,3 +230,928 @@ XCM
...
+entrancy
+enviroment
+envolved
+epoc
```
Comment (benluelo) [inline .config/spellcheck.dic:482]:
```suggestion
```

File: .config/spellcheck.dic [hunk 21]
```diff
@@ -230,3 +230,928 @@ XCM
...
+EQLC
+eququal
```
Comment (benluelo) [inline .config/spellcheck.dic:484]:
```suggestion
```

File: .config/spellcheck.dic [hunk 22]
```diff
@@ -230,3 +230,928 @@ XCM
...
+ETHBTC
+Etherium
+evaluatable
+everytime
+excempt
+exectuor
+exhastive
+existant
+expireable
```
Comment (benluelo) [inline .config/spellcheck.dic:493]:
```suggestion
```

File: .config/spellcheck.dic [hunk 23]
```diff
@@ -230,3 +230,928 @@ XCM
...
+extrsinsic
+failes
+feautes
+feeconfig
+feesless
+Fetador
+Finalised
+findutils
+fintech
+fitlered
```
Comment (benluelo) [inline .config/spellcheck.dic:508]:
```suggestion
```

File: .config/spellcheck.dic [hunk 24]
```diff
@@ -230,3 +230,928 @@ XCM
...
+flashloans
+fnft
+Fnft
+FNFT
+followin
```
Comment (benluelo) [inline .config/spellcheck.dic:513]:
```suggestion
```

File: .config/spellcheck.dic [hunk 25]
```diff
@@ -230,3 +230,928 @@ XCM
...
+fontconfig
+FRAX
+freetype
+Fridman
+frontrunable
+frop
+FSIZE
+fullfilled
+funcitonal
+functionl
```
Comment (benluelo) [inline .config/spellcheck.dic:523]:
```suggestion
```

**Review by benluelo — COMMENTED** [review]
only got to G, lots of these aren't words

**Comment by andor0** [comment]
> only got to G, lots of these aren't words

perhaps, we should fix spellcheck check in a separate PR.

File: .config/spellcheck.dic [hunk 26]
```diff
@@ -230,3 +230,928 @@ XCM
...
+westend
+Westend
+Whenwe
+whos
+Wickramasinghe
+withdrawed
+withdrawl
+wococo
+Wococo
+writting
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:1135]: writing

File: .config/spellcheck.dic [hunk 27]
```diff
@@ -230,3 +230,928 @@ XCM
...
+wasmtime
+wasmvm
+webassembly
+websocat
+websockat
+Wehbi
+weigth
+westend
+Westend
+Whenwe
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:1128]: When we

File: .config/spellcheck.dic [hunk 28]
```diff
@@ -230,3 +230,928 @@ XCM
...
+tendermint
+Tendermint
+tendermint's
+Tentang
+testgen
+TGEs
+theupdate
+thiserror
+thoose
+threre
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:1015]: there

File: .config/spellcheck.dic [hunk 29]
```diff
@@ -230,3 +230,928 @@ XCM
...
+squeezy
+stablecoins
+Stablecoins
+stackable
+stackalt
+stakedb
+stakerclaimunstake
+stakers
+Stakers
+stakersn
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:964]: stakers

File: .config/spellcheck.dic [hunk 30]
```diff
@@ -230,3 +230,928 @@ XCM
...
+selectables
+Selland
+seqs
+serie
+Serie
+sersice
+Seun
+shareholdres
+shold
+Siddiqui
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:935]: Subsquid

File: .config/spellcheck.dic [hunk 31]
```diff
@@ -230,3 +230,928 @@ XCM
...
+pseudocode
+pubkey
+Puchased
+pyth
+Pyth
+PYTH
+pythd
+PYTHD
+quickiswap
+Quuery
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:846]: Query

File: .config/spellcheck.dic [hunk 32]
```diff
@@ -230,3 +230,928 @@ XCM
...
+millify
+Millisecs
+MILLISECS
+mimimal
+minial
+Minimalistic
+minterd
+mintues
+minvested
+miterlimit
```
Comment (dzmitry-lahoda) [inline .config/spellcheck.dic:672]: minterlimit

**Comment by benluelo** [comment]
> > only got to G, lots of these aren't words
> 
> perhaps, we should fix spellcheck check in a separate PR.

I think it would be good to fix this now - I can help fix the spelling errors, if we all take a letter range I think it should be fairly doable (I can A-G)

File: .config/spellcheck.dic [hunk 33]
```diff
@@ -230,3 +230,928 @@ XCM
...
+concat
+conect
+configuraiton
+configuraitons
+Confing
+Conntected
+Consant
+consenssus
+Consensys
+consta
```
[the remark this answers is not in the document]
Comment (benluelo) [reply .config/spellcheck.dic:390]: Consensys is a company, apparently

File: .config/spellcheck.dic (hunk 18 above)
Comment (benluelo) [reply .config/spellcheck.dic:473]: emphasising is british spelling so i guess it's valid

File: .config/spellcheck.dic [hunk 34]
```diff
@@ -230,3 +230,928 @@ XCM
...
+funcitonal
+functionl
+Fundsto
+Gauba
+GENERALISED
+Genrates
+gnugrep
+gnumake
+graphviz
+Guarranteed
```
Comment (benluelo) [inline .config/spellcheck.dic:531]:
```suggestion
```

File: .config/spellcheck.dic [hunk 35]
```diff
@@ -230,3 +230,928 @@ XCM
...
+rebalance
+rebalances
+rebalancing
+Rebalancing
+recepients
+reciepts
+recieve
+recieved
+recognised
+Recvs
```
[the remark this answers is not in the document]
Comment (andor0) [reply .config/spellcheck.dic:806]: looks like acronyms of `Receivers`

File: .config/spellcheck.dic (hunk 32 above)
Comment (andor0) [reply .config/spellcheck.dic:614]: it uses in *.svg
it is a [css attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit)

File: .config/spellcheck.dic (hunk 31 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic (hunk 30 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic (hunk 29 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic [hunk 36]
```diff
@@ -230,3 +230,928 @@ XCM
...
+strikethrough
+STRM
+structopt
+Styleguide
+subaccount
+subcommmand
+subkeys
+Submittables
+Substate
+substential
```
[the remark this answers is not in the document]
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic (hunk 28 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic (hunk 27 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic [hunk 37]
```diff
@@ -230,3 +230,928 @@ XCM
...
+websocat
+websockat
+Wehbi
+weigth
+westend
+Westend
+Whenwe
+whos
+Wickramasinghe
+withdrawed
```
[the remark this answers is not in the document]
Comment (andor0) [reply .config/spellcheck.dic]: fixed

File: .config/spellcheck.dic (hunk 26 above)
Comment (andor0) [reply .config/spellcheck.dic]: fixed

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2wp5e0r Add tests with coverage and fix cspell warnings in nix CI pipeline](https://app.clickup.com/t/2wp5e0r)

**Review by easteregg — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Comment by andor0** [comment]
@Mergifyio refresh

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]
also i think that editor oriented dic naming is better not domain oriented. fine word repeat i. both dictionaries. now you have to decide. ow youtlr own identity but word domain which is ambiguilt and hard

**Comment by andor0** [comment]
@Mergifyio refresh

**Comment by andor0** [comment]
@Mergifyio refresh

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by Dom-Roth — APPROVED** [review]

**Comment by andor0** [comment]
@Mergifyio refresh

**Review by benluelo — APPROVED** [review]

**Comment by andor0** [comment]
@Mergifyio refresh

**Comment by hussein-aitlahcen** [comment]
@mergifyio refresh

**Comment by hussein-aitlahcen** [comment]
@mergifyio update

**Comment by hussein-aitlahcen** [comment]
@mergifyio queue

**Comment by hussein-aitlahcen** [comment]
@mergifyio requeue

**Comment by hussein-aitlahcen** [comment]
@mergifyio refresh

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by alexkdefi — APPROVED** [review]
👍

**Comment by andor0** [comment]
@Mergifyio refresh

**Review by 0xBrainjar2 — APPROVED** [review]

**Comment by andor0** [comment]
@Mergifyio refresh

**Pull request merged**